addons/product_images/wizard/product_get_pic_using_barcode.py

```diff
@@ -0,0 +1,219 @@
+            self._get_pics_from_google(self.product_ids[:10])
+        except HTTPError as http_error:
+            if http_error.response.status_code == requests.codes.too_many_requests:
+                self.env.ref('product_images.ir_cron_get_pics_from_google')._trigger(
+                    fields.datetime.now() + timedelta(days=1.0)
+                )
+                return {
+                    'type': 'ir.actions.client',
+                    'tag': 'display_notification',
+                    'params': {
+                        'type': 'warning',
+                        'title': _("Product images"),
+                        'message': _("Custom search quota exceeded. We will start again tomorrow"),
+                        'next': {'type': 'ir.actions.act_window_close'},
+                    }
+                }
+            elif http_error.response.status_code == requests.codes.forbidden:
+                project_url = re.search(r"(?P<url>https?://[^\s]+)",
+                            http_error.response.json().get('error').get('message')).group('url')
+                raise UserError(_("Custom Search API is not enabled in your Google project. "
+                                  "Enable it by visiting %(url)s then retry. If you enabled this "
+                                  "API recently, wait a few minutes for the action to propagate "
+                                  "to Google's systems and retry.", url=project_url))
+            elif http_error.response.status_code == requests.codes.bad_request:
+                raise UserError(_("Your API Key or your Search Engine ID is incorrect."))
+        except ConnectionError:
+            self.env.ref('product_images.ir_cron_get_pics_from_google')._trigger(
+                fields.datetime.now() + timedelta(hours=1.0)
+            )
+        else:
+            if self.product_ids[10:]:
+                self.env.ref('product_images.ir_cron_get_pics_from_google')._trigger()
+                return {
+                    'type': 'ir.actions.client',
+                    'tag': 'display_notification',
+                    'params': {
+                        'type': 'success',
+                        'title': _("Product images"),
+                        'message': _("We are processing your products. Please come back in a few "
+                                    "moments to see the images."),
+                        'next': {'type': 'ir.actions.act_window_close'},
+                    }
+                }
+
+    def get_pics_using_barcode(self):
+        """ Get pictures to a list of product using the barcode
+
+        This function is called from a cron job. If the daily request limit is reach, the cron job
+        will be schedule to run again a day later.
+        """
+        product_ids = self.env['product.product'].search([
+            ('to_be_processed_for_image', '=', 'True'),
+        ])
+        try:
+            self._get_pics_from_google(product_ids[:100])
```

Comment:
It's a great improvement of my code but, I wonder, how do I know if I need to restart a cron  ? If I need to do a second search it may not be worth it .

Reply:
I didn't though of that. I'll make the changes ASAP